### PR TITLE
Add unit tests for formatter json

### PR DESCRIPTION
**Pull request by VVeremjova**
Added tests for converting smart object with response and error response
types into string

Relates: APPLINK-20090

### CODE REVIEW

**Comment by VVeremjova** [comment]
please review : @malirod, @AByzhynar, @Kozoriz, @OHerasym

File: src/components/formatters/test/formatter_json_rpc_test.cc [hunk 1]
```diff
@@ -30,9 +30,9 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <algorithm>
-#include "gtest/gtest.h"
 #include "formatters/formatter_json_rpc.h"
+#include "string.h"
```
Comment (Kozoriz) [inline src/components/formatters/test/formatter_json_rpc_test.cc]: <string>

File: src/components/formatters/test/formatter_json_rpc_test.cc [hunk 2]
```diff
@@ -79,15 +79,15 @@ TEST(FormatterJsonRPCTest, CorrectRPCv2_request_SmartObjectToString_EXPECT_SUCCE
...
-          "{\n   \"id\" : 4444,\n   \"jsonrpc\" : \"2.0\",\n   \"method\" : \"AddCommandID\"\n}\n"),
-      result);
+  EXPECT_EQ(std::string(
+                "{\n   \"id\" : 4444,\n   \"jsonrpc\" : \"2.0\",\n   "
+                "\"method\" : \"AddCommandID\"\n}\n"),
+            result);
 }
 
-TEST(FormatterJsonRPCTest, CorrectRPCv1_notification_SmartObjectToString_EXPECT_SUCCESS) {
+TEST(FormatterJsonRPCTest, CorrectRPCv1Notification_ToString_EXPECT_SUCCESS) {
```
Comment (Kozoriz) [inline src/components/formatters/test/formatter_json_rpc_test.cc]: don`t write EXPECT in naming

**Comment by OHerasym** [comment]
Reviewed

**Comment by malirod** [comment]
Reviewed

**Comment by Kozoriz** [comment]
reviewed

**Comment by AByzhynar** [comment]
Reviewed

**Pull request merged**